music_assistant/providers/pandora/provider.py

```diff
@@ -0,0 +1,454 @@
+                queue_item = self.mass.player_queues.get_item(queue_id, queue_item_id)
+                if queue_item and queue_item.streamdetails:
+                    # Get the best quality album art
+                    album_art_url = None
+                    if album_art := track.get("albumArt"):
+                        # Get the 500px version (good balance of quality/size)
+                        album_art_url = next(
+                            (art["url"] for art in album_art if art.get("size") == 500),
+                            album_art[-1]["url"] if album_art else None,
+                        )
+                    queue_item.streamdetails.stream_metadata = StreamMetadata(
+                        title=track.get("songTitle", "Unknown Song"),
+                        artist=track.get("artistName", "Unknown Artist"),
+                        album=track.get("albumTitle"),
+                        image_url=album_art_url,
+                        duration=track.get("trackLength"),
+                    )
+                    self.logger.debug(
+                        "Updated stream metadata: %s - %s",
+                        track.get("artistName", "Unknown"),
+                        track.get("songTitle", "Unknown"),
+                    )
+
+            self.logger.debug(
+                "Redirecting to audio URL for track %d (%s - %s)",
+                track_num,
+                track.get("artistName", "Unknown"),
+                track.get("songTitle", "Unknown"),
+            )
+
+            # Redirect to the actual audio URL
+            return web.Response(status=302, headers={"Location": audio_url})
+
+        except Exception as err:
+            self.logger.error(
+                "Error handling stream request for track %d: %s",
+                track_num,
+                err,
+            )
+            return web.Response(status=500, text="A stream error occurred")
+
+    async def search(
+        self,
+        search_query: str,
+        media_types: list[MediaType],
+        limit: int = 25,
+    ) -> SearchResults:
+        """Search library radio stations by name."""
+        # We can't search Pandora's API without the legacy endpoints,
+        # but we can search the user's existing library stations
+        if MediaType.RADIO not in media_types:
+            return SearchResults()
+
+        results: list[Radio] = []
+
+        async for station in self.get_library_radios():
+            if compare_strings(station.name, search_query):
+                results.append(station)
+                if len(results) >= limit:
+                    break
+
+        return SearchResults(radio=results)
+
+    async def browse(self, path: str) -> list[Radio]:
+        """Browse radio stations."""
+        # For now, just return all stations like get_library_radios
+        # Could be enhanced with categories/genres in the future
+        return [station async for station in self.get_library_radios()]
```

Comment:
```suggestion
```

Reply:
I am hearing you and I don't know why but if I remove that and then remove and re-add the provider then the browse folder is empty. I'm sure I'm doing something wrong but I don't know what!

<img width="472" height="263" alt="Image" src="https://github.com/user-attachments/assets/45c4a59d-34f9-4f3d-b9de-d510600f4ee8" />

and then with that function put back

<img width="463" height="345" alt="Image" src="https://github.com/user-attachments/assets/7760e5a5-aa91-4fae-80ec-3310f363bc52" />